### ISSUE TITLE
Collection View header binding

### ISSUE THREAD

**Issue by ghaith100994** [xamarin/Xamarin.Forms#8388]
## Summary
Can add a label to the header of collection view and bind the text of the header

## API Changes

`<collectionview >
<collectionview.header>
<Label text="{binding title}"/>
</collectionview.header>
</collectionview>`

**Comment by hartez** [comment]
You can already do this, you just need to use the HeaderTemplate property:

```
<CollectionView x:Name="CollectionView" Header="{Binding .}" ItemsSource="{Binding Items}">
	<CollectionView.HeaderTemplate>
		<DataTemplate>
				<Label Text="{Binding Title}" />
		</DataTemplate>
	</CollectionView.HeaderTemplate>
</CollectionView>
```

You can see a full example [here](https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/HeaderFooterGalleries/HeaderFooterTemplate.xaml) and [here](https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.Controls/GalleryPages/CollectionViewGalleries/HeaderFooterGalleries/HeaderFooterTemplate.xaml.cs).

**Comment by ghaith100994** [comment]
@hartez  I did it like you said and it did not work after changing the binding

`
<CollectionView x:Name="MenuItemsListView"
                    SelectionMode="Single"
                    Header="{Binding .}"
                    SelectedItem="{Binding SelectedChannel}"
                    ItemsSource="{Binding ChannelList}">
            <CollectionView.ItemsLayout>
                <LinearItemsLayout Orientation="Vertical" ItemSpacing="0" />
            </CollectionView.ItemsLayout>
            <CollectionView.HeaderTemplate>
                <DataTemplate>
                <Grid >
                    <Grid.ColumnDefinitions>
                        <ColumnDefinition Width="10"/>
                        <ColumnDefinition Width="*"/>
                    </Grid.ColumnDefinitions>
                    <Grid.RowDefinitions>
                            <RowDefinition Height="Auto"/>
                            <RowDefinition Height="Auto"/>
                            <RowDefinition Height="10"/>
                    </Grid.RowDefinitions>
                        <Grid HeightRequest="90">
                            <ffimageloading:CachedImage Aspect="AspectFit" 
                                        Grid.Row="0" 
                                        Margin="65,5"
                                        Source ="{Binding LogoBackgroundImage}"/>
                        </Grid>
                        <Label Grid.Column="1"
                               Text="Live Channels"
                               FontSize="20"
                               Grid.Row="1"
                               VerticalOptions="Start"
                               HorizontalOptions="Start"
                               FontFamily="{StaticResource CairoBold}" 
                               TextColor="White"
                               Style="{DynamicResource SubtitleStyle}"/>
                </Grid>
                </DataTemplate>
            </CollectionView.HeaderTemplate>
            <CollectionView.ItemTemplate>
                <DataTemplate>
                    <Grid HeightRequest="58">
                        <Grid.ColumnDefinitions>
                            <ColumnDefinition Width="Auto"/>
                            <ColumnDefinition Width="60"/>
                            <ColumnDefinition Width="*"/>
                        </Grid.ColumnDefinitions>
                        <Label Margin="10,5" FontFamily="{StaticResource CairoSemiBold}"  Text="{Binding FixedNumber}" Grid.Column="0"  TextColor="White" HorizontalOptions="Center" VerticalOptions="Center" FontSize="16" />
                        <ffimageloading:CachedImage Margin="5,10"
                                                    Grid.Column="1"
                                                    CacheType="Memory"
                                                    Source = "{Binding IconUrl}"/>
                        <Label Text="{Binding Name}" FontFamily="{StaticResource CairoSemiBold}"  Grid.Column="2" Margin="5" TextColor="White" HorizontalOptions="Start" VerticalOptions="Center"  FontSize="16"/>
                        <Grid.Style>
                            <Style TargetType="Grid">
                                <Setter Property="VisualStateManager.VisualStateGroups">
                                    <VisualStateGroupList >
                                        <VisualStateGroup x:Name="CommonStates">
                                            <VisualState x:Name="Normal" />
                                            <VisualState x:Name="Selected">
                                                <VisualState.Setters>
                                                    <Setter Property="BackgroundColor" Value="#5b2e8d"/>

                                                </VisualState.Setters>
                                            </VisualState>
                                        </VisualStateGroup>
                                    </VisualStateGroupList>
                                </Setter>
                            </Style>
                        </Grid.Style>
                    </Grid>
                </DataTemplate>
             </CollectionView.ItemTemplate>
        </CollectionView>
`
While When using List view It Work

**Comment by ghaith100994** [comment]
sorry it's work 
it is my error

**Comment by jsauve** [comment]
FYI, the API is a bit confusing as-is. If one specifies bindable elements as children of <CollectionView.Header>, they render as expected on iOS. Those same bindable elements on Android will render their layout areas, but will not bind. This had me stumped for a couple days until I stumbled upon this issue thread and found that I needed to instead specify my bindable elements as children of <CollectionView.HeaderTemplate>, and then set the header binding like so: <CollectionView Header="{Binding .}">. Having the API behave in two different ways is confusing.
